SIG Contribex 2021 Annual Report (#6411)
kubernetes/community: sig-contributor-experience/annual-report-2021.md

1. What work did the SIG do this year that should be highlighted?

   - [Migrating K-Dev to a managed kubernetes.io account](https://github.com/kubernetes/community/issues/5877)

>    - Steering Election with Elekto

Reviewer: describe elekto a little more - created by lfx intern, what problem did it solve? etc

      - Major change to how we do elections

>       - There were some hiccups but overall it's a step in the right direction

Reviewer: I don't think you need this line unless you go into what hiccups there were and why its important to mention that here

>    - Community meeting overhaul

Reviewer: ```suggestion
   - Community meeting overhaul and reboot
```

>    - EasyCLA migration

Reviewer: what was the impact of this? across how many repos?

Reviewer 2: All orgs, 300+ repos, roughly 7.5 active committers

   - Contributor Summit north america
   - Contributor Celebration
   - Restaffed up teams
      - Moderation

>       - Youtube Admin

Reviewer: also - curated and ran 8 months of Leads meetings (16 meetings total)

  |         Subproject        |          Initiative / Program          |
  |:-------------------------:|:--------------------------------------:|
  | Community                 | Community Repo Stewardship             |
  | Community Management      | Annual Contributor Survey              |
  | Community Management      | Calendar Admin                         |
  | Community Management      | Chair and TL Meetings + Docs           |
  | Community Management      | Discuss Admin                          |
  | Community Management      | Mailing List Admin                     |
  | Community Management      | Slack Admin                            |
  | Community Management      | Zoom / YouTube Admin                   |
  | Contributor Documentation | Contributor Guide Stewardship          |
  | Contributor Documentation | Contributor Site                       |
  | Contributor Documentation | Developer Guide Audit                  |
  | Contributor Documentation | Developer Guide Stewardship            |
  | Contributors Comms        | Contributor / SIG Profiling            |
  | Contributors Comms        | Stewardship of k8scontributors twitter |

>   | Devstats                  | Devstats Dashboard Update              |

Reviewer 2: sunsetting which one? - in general though I think things should be included if they were activities we did last year. We could put a note in there though.

>   | Devstats                  | Devstats Dashboard Update              |

Author: If this is Devstats I'm unsure if we should say we are considering sunsetting at this time.

I think it's best we leave this as it is.

What do you think @mrbobbytables ?

>   | Devstats                  | Devstats Dashboard Update              |

Reviewer 2: I think we leave as is and remove it for 2022 👍

  | Events                    | Monthly Community Meeting              |

>   | Events                    | Office Hours                           |

Author: I think it should be included if we did it last year, and agree with the side notes on making comments about sunsetting.

Will make some changes.

>   | Events                    | Office Hours                           |

Reviewer 2: Yeah, +1 include it as we did it last year. We can say we're looking into sunsetting it, but I'd avoid committing to sunsetting till we actually do it

>   | Events                    | Elections                              |

Author: Fixed with table update

  | GitHub Management         | GitHub Admin / Moderation              |
  | GitHub Management         | GitHub Master -> Main rename           |
  | GitHub Management         | GitHub New Membership Coordinator      |
  | Mentoring                 | Google Summer of Code                  |
  | Mentoring                 | Group Mentoring                        |
  | Mentoring                 | LFX Mentor Program                     |
  | Mentoring                 | Meet our Contributors                  |

>   | Mentoring                 | Outreachy                              |

Reviewer: ```suggestion
  | Mentoring                 | Outreachy                              |
  | Mentoring                 | New Contributor Workshop  |
```

Reviewer 2: We didn't really do anything with the NCW last year

Reviewer: I read this as things we are working on currently

Reviewer: it would be cool if this list had issues and things attached to it like what it is we are working on in each area and maybe a vision statement for each. I can help with this.

Author: added! Thank you @mrbobbytables for putting the table together 🙏

  | Slack Infra               | slack-infra                            |

## Project health

1. What areas and/or subprojects does your group need the most help with?
   Any areas with 2 or fewer OWNERs? (link to more details)
   
   - GitHub Admin Subproject
      - Looking to bring on more membership co-ordinators 
   - Community Management Automation
      - [Zoom to Youtube](https://github.com/kubernetes/community/issues/5201)
      - [Workspace Automation](https://github.com/kubernetes/steering/issues/213)
         - Mailing list management
         - Calendar management
         - Shared drives
      - [Mailing list migration](https://github.com/kubernetes/community/issues/6353)

> 

Author: Mentoring needs help, Having difficulty rescheduling the mentoring meeting,
Co-ordinating with outreachy, LFX & running mentoring cohorts.

2. What metrics/community health stats does your group care about and/or measure?

   - Issue Velocity

>       - We are a service orientated sig, Issues are the main form of how we track work.

Author: TODO - Link to issue velocity dashboard for Contribex related issues

3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
   to activities or programs that provide useful context or allow easy participation?

   - We don't have a CONTRIBUTING.md as our SIG's contributing proccess does not differ from the proccess in the Contributor Guide.

4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
   does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?

   - N/A

5. Does the group have contributors from multiple companies/affiliations?

   - Yes. [25+ Different groups](https://k8s.devstats.cncf.io/d/8/company-statistics-by-repository-group?orgId=1&var-period=d7&var-metric=contributions&var-repogroup_name=SIG%20Contributor%20Experience&var-repo_name=kubernetes%2Fkubernetes&var-companies=All&from=1577854800000&to=1640926800000&viewPanel=1)

6. Are there ways end users/companies can contribute that they currently are not?
   If one of those ways is more full time support, what would they work on and why?

>    - More full time support for the items under bullet point 1.

Reviewer: A full time community manager who can maintain several subprojects and help with the overall direction of the group

    - This work will reduce toil for not only contributor experience, but all groups in the project.

## Membership

> Statistics were retrived on 2021-01-30

Author: 2022 not 2021

| [Mentoring](https://git.k8s.io/community/sig-contributor-experience#mentoring)                                  | [LFX Mentor Program](https://git.k8s.io/community/mentoring/programs/lfx-mentoring.md)                                |
| [Slack Infra](https://git.k8s.io/community/sig-contributor-experience#slack-infra)                              | [slack-infra](https://sigs.k8s.io/slack-infra)                                                                        |

  We put the following initiatives on hold, and will be revisiting these at a later date.

  |         Subproject        |          Initiative / Program          |
  |:-------------------------:|:--------------------------------------:|
  | Mentoring                 | Meet our Contributors                  |
  | Mentoring                 | Outreachy                              |
  | Mentoring                 | New Contributor Workshop               |
  | Mentoring                 | Google Summer of Code                  |

## Project health

1. What areas and/or subprojects does your group need the most help with?
   Any areas with 2 or fewer OWNERs? (link to more details)
   
   - GitHub Admin Subproject

>       - We need to bring on more [new membership co-ordinators](https://cs.github.com/kubernetes/community/blob/cf77df8935eb5460d8d2856c40009d981691ec2f/github-management/README.md#L49).

Reviewer 2: ```suggestion
      - We need to bring on more [new membership coordinators](https://git.k8s.io/community/github-management#new-membership-coordinator).
```

just caught this, we should not link to code search but link to it directly
